Add Swift 3.0 support.

nlutsenko opened the pull request:
- Updating all source files to Swift 3.0 syntax
- Updating SwiftCatalog to Swift 3.0 and iOS 10

Most of the changes are simply due to language syntax change, but there are few API changes as well to make the SDK more Swifty :)

Kudos and thanks to:
- @gerbiljames for the initial request in #55 that gets superseded now.
- @milseman for clarifying `@escaping` behavior with generic types (it's implicitly `@escaping`)

Closes #53

richardjrossiii commented on Sources/Core/AppEvents/AppEvent.Builtin.swift:
```diff
@@ -49,12 +49,12 @@ extension AppEvent {
-  public static func CompletedTutorial(successful successful: BooleanType? = nil,
-                                                  valueToSum: Double? = nil,
-                                                  extraParameters: ParametersDictionary = [:]) -> AppEvent {
+  public static func completedTutorial(successful: DarwinBoolean? = nil,
```
`DarwinBoolean` probably isn't the right choice here.

nlutsenko replied:
Good catch, need to update.

richardjrossiii commented on Sources/Core/AppEvents/AppEvent.Builtin.swift:
```diff
@@ -91,16 +91,16 @@ extension AppEvent {
-                                        extraParameters: ParametersDictionary = [:]) -> AppEvent {
+  public static func searched(contentId: String? = nil,
+                              searchedString: String? = nil,
+                              successful: DarwinBoolean? = nil,
```
`DarwinBoolean`

nlutsenko replied:
Good catch, need to update.

richardjrossiii commented on Sources/Core/AppEvents/AppEvent.Builtin.swift:
```diff
@@ -206,12 +205,12 @@ extension AppEvent {
-  public static func AddedPaymentInfo(successful successful: BooleanType? = nil,
-                                                 valueToSum: Double? = nil,
-                                                 extraParameters: ParametersDictionary = [:]) -> AppEvent {
+  public static func addedPaymentInfo(successful: DarwinBoolean? = nil,
```
`DarwinBoolean`

nlutsenko replied:
Good catch, need to update.

richardjrossiii commented on Sources/Core/GraphRequest/GraphRequestProtocol.swift:
```diff
@@ -59,7 +59,7 @@ extension GraphRequestProtocol {
    - parameter completion: Optional completion closure that is going to be called when the connection finishes or fails.
    */
-  public func start(completion: ((httpResponse: NSHTTPURLResponse?, result: GraphRequestResult<Self>) -> Void)? = nil) {
+  public func start(_ completion: @escaping GraphRequestConnection.Completion<Self>) {
```
Shouldn't this be `GraphRequestConnection<Self>.Completion`? The typealias itself doesn't have the generic paramter, `GraphRequestConnection` does, no?

nlutsenko replied:
Yes, will update.

richardjrossiii commented on Sources/Core/UserProfile/UserProfile.PictureView.swift:
```diff
@@ -24,12 +24,12 @@ import FBSDKCoreKit.FBSDKProfilePictureView
 
     /// The aspect ratio of the source image of the profile picture.
-    public var pictureAspectRatio = UserProfile.PictureAspectRatio.Square {
+    open var pictureAspectRatio = UserProfile.PictureAspectRatio.square {
```
`open` here should do nothing, as `PictureView` is final, no?

nlutsenko replied:
I knew I missed one of these.

richardjrossiii commented on Sources/Login/LoginManager.swift:
```diff
@@ -98,31 +98,31 @@ public class LoginManager {
    This calls `AccessToken.current = nil` and `Profile.current = nil`.
    */
-  public func logOut() {
+  open func logOut() {
```
Again, `open` on a final class?

nlutsenko replied:
Yup, should be `public`.

nlutsenko commented:
@richardjrossiii, updated and rebased:
- Fixed all violations from `open` to `public`
- Replaced `DarwinBoolean` with `Bool`
- Made `completion` optional again on `GraphRequestProtocol.start()`.

aliabadi commented:
When I add the Facebook dependencies to my Podfile it loads an older version of the SDK.

Is there something I'm missing?  The readme says that's all you should have to do.  I've tried copying all the newer swift files in, but that was fraught with errors...I was able to get it working by using the legacy swift option.

Thanks!

ricardochavarria commented:
This seems to be merged into *master*, but I can't get it to compile with Swift 3.0 in Xcode 8.0.

simonstewart commented:
@ricardochavarria I'm seeing this issue as of today.  Have you worked around it or still waiting?

aliabadi commented:
I did find a workaround of sorts.   I load the standard SDK through my podfile and when I open the new workspace I run the "Convert to Swift 3 Syntax" wizard and it works after that.  

Just have to build the project afterwards and everything is fine.

Thanks!

idelfonsog2 commented:
@aliabadi when you said the standard... Do you mean the Obj-C or the swift sdk?
